### ISSUE TITLE
Patch To Embedding correct?

### ISSUE THREAD

**Issue by tahmid0007** [lucidrains/vit-pytorch#14]
In line 95 of ViT [[self.patch_to_embedding = nn.Linear(patch_dim, dim)]]
Is it supposed to be a nn.Linear layer? I believe its a learnable tensor. The paper says "E is a trainable linear projection that maps each vectorized patch to the model dimension D". Yannic also referred E as a linear projection matrix. Could you please share your thoughts?

NB: I have modified and run E as a nn.parameter type tensor and produces similar results

**Comment by lucidrains** [comment]
@tahmid0007 I believe it is the same?

**Comment by tahmid0007** [comment]
@lucidrains I understand your point of view. If you figure out any difference, please let us know. I appreciate your work :) .

**Comment by lucidrains** [comment]
Thanks! I don't believe there is a difference, so I'll close the issue unless you (or someone else) can present evidence otherwise
